test/v3/services/CreatePeriodSummaryServiceSpec.scala

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v3.services
+
+import anyVersion.models.request.createPeriodSummary.{PeriodDates, PeriodDisallowableExpenses, PeriodIncome}
+import anyVersion.models.response.createPeriodSummary.CreatePeriodSummaryResponse
+import api.controllers.EndpointLogContext
+import api.models.domain.{BusinessId, Nino}
+import api.models.errors._
+import api.models.outcomes.ResponseWrapper
+import api.services.ServiceSpec
+import v3.mocks.connectors.MockCreatePeriodSummaryConnector
+import v3.models.request.createPeriodSummary._
+
+import scala.concurrent.Future
+
+class CreatePeriodSummaryServiceSpec extends ServiceSpec {
+
+  val nino: String                   = "AA123456A"
+  val businessId: String             = "XAIS12345678910"
+  implicit val correlationId: String = "X-123"
+
+  private val requestBody: CreatePeriodSummaryBody =
+    CreatePeriodSummaryBody(
+      PeriodDates("2019-08-24", "2019-08-24"),
```

Comment:
Same as in the connector spec - we can set the optionals to none as the data isnt the focus of the test which will make this file easier to read and the tests intent clearer